### PR TITLE
Optimism - Add Chainlink Oracles

### PR DESCRIPTION
**Pull request by MSilb7**
Brief comments on the purpose of your changes:
Add Chainlink oracle updates for Optimism.
Follow the design pattern from Uniswap that allows for eventual aggregation across chains.

*For Dune Engine V2*
I've checked that:
General checks:
* [x] I tested the query on dune.com after compiling the model with dbt compile (compiled queries are written to the target directory)
* [x] I used "refs" to reference other models in this repo and "sources" to reference raw or decoded tables 
* [ ] if adding a new model, I added a test
* [x] the filename is unique and ends with .sql
* [x] each sql file is a select statement and has only one view, table or function defined  
* [x] column names are `lowercase_snake_cased`
* [x] if adding a new model, I edited the dbt project YAML file with new directory path for both models and seeds (if applicable)
* [x] if wanting to expose a model in the UI (Dune data explorer), I added a post-hook in the JINJA config to add metadata (blockchains, sector/project, name and contributors)

Join logic:
* [ ] if joining to base table (i.e. ethereum transactions or traces), I looked to make it an inner join if possible

Incremental logic:
* [ ] I used is_incremental & not is_incremental jinja block filters on both base tables and decoded tables
  * [ ] where block_time >= date_trunc("day", now() - interval '1 week')
* [ ] if joining to base table (i.e. ethereum transactions or traces), I applied join condition where block_time >= date_trunc("day", now() - interval '1 week')
* [ ] if joining to prices view, I applied join condition where minute >= date_trunc("day", now() - interval '1 week')

### CODE REVIEW

**Comment by MSilb7** [comment]
^ whew, tried to clear out some other changes that accidentally got saved here. Should be good to go now, assuming no other dbt compile issues.

**Comment by MSilb7** [comment]
Finally 🙌

Had some duplication in the raw table, have a test for it now.

File: models/chainlink/optimism/chainlink_optimism_sources.yml [hunk 1]
```diff
@@ -0,0 +1,11 @@
+version: 2
+
+sources:
+  - name: coptimism
```
[the remark this answers is not in the document]
Comment (MSilb7) [reply models/chainlink/optimism/chainlink_optimism_sources.yml]: good call fixed

File: models/chainlink/chainlink_price_feeds_schema.yml [hunk 2]
```diff
@@ -0,0 +1,49 @@
...
+    tests:
+      - dbt_utils.unique_combination_of_columns:
+          combination_of_columns:
+              - blockchain
+              - hour
+              - proxy_address
+              - aggregator_address
+              - underlying_token_address
+    columns:
+      - &blockchain_all
```
[the remark this answers is not in the document]
Comment (MSilb7) [reply models/chainlink/chainlink_price_feeds_schema.yml:23]: i just copied other tables, no idea why this is how it is

File: models/chainlink/optimism/chainlink_optimism_schema.yml [hunk 3]
```diff
@@ -0,0 +1,151 @@
...
+        Table pulling price updates for known Chainlink oracle addresses, truncated to hourly average price for easier joining with other on-chain tables.
+    tests:
+      - dbt_utils.unique_combination_of_columns:
+          combination_of_columns:
+              - blockchain
+              - hour
+              - proxy_address
+              - aggregator_address
+              - underlying_token_address
+    columns:
```
[the remark this answers is not in the document]
Comment (MSilb7) [reply models/chainlink/optimism/chainlink_optimism_schema.yml:67]: got it, i'll update

File: models/chainlink/chainlink_price_feeds_schema.yml [hunk 4]
```diff
@@ -0,0 +1,50 @@
...
```
[the remark this answers is not in the document]
Comment (MSilb7) [reply models/chainlink/chainlink_price_feeds_schema.yml:23]: just trying to differentiate the naming vs the prior schema, but i'll remove it

File: models/chainlink/chainlink_price_feeds_schema.yml [hunk 5]
```diff
@@ -0,0 +1,49 @@
...
+      blockchain: optimism
+      sector: oracle
+      project: chainlink
+      contributors: msilb7
+    config:
+      tags: ['optimism', 'chainlink','oracle']
+    description: >
+        Hourly chainlink price feed updates on all chains across all contracts and versions.
+    tests:
+      - dbt_utils.unique_combination_of_columns:
```
[the remark this answers is not in the document]
Comment (jeff-dude) [reply models/chainlink/chainlink_price_feeds_schema.yml:15]: yes 🙏

**Comment by MSilb7** [comment]
I'm not getting the duplication when I run it, but in case it's caused by the incremental merge, I forced the underlying token address to 'n/a' if there's no match.

**Review by jeff-dude — APPROVED** [review]
thank you, should get deployed soon

**Pull request merged**